src/ol/ol.js

```diff
@@ -297,3 +311,48 @@ if (typeof window !== 'undefined') {
 } else if (typeof self !== 'undefined') {
   ol.global = self;
 }
+
+
+/**
+ * Error object thrown when an assertion failed. This is an ECMA-262 Error,
+ * extended with a `code` property.
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error}
+ * @constructor
+ * @extends {Error}
+ * @implements {oli.AssertionError}
+ * @param {number} code Error code.
+ */
+ol.AssertionError = function(code) {
+
+  /**
+   * @type {string}
+   */
+  this.message = 'Assertion failed. See ' +
+      (ol.VERSION ? 'http://openlayers.org/en/' + ol.VERSION.split('-')[0] : '') +
+      '/doc/errors.html#' + code + ' for details.';
```

Comment:
URL snob comment: how about `openlayers.org/en/latest/errors/#123` instead (adding `doc/errors/index.md` instead of `doc/errors.md`?